lib/web/apiserver_test.go

```diff
@@ -2393,6 +2401,91 @@ func TestWebSessionsRenewAllowsOldBearerTokenToLinger(t *testing.T) {
 	require.True(t, trace.IsAccessDenied(err))
 }
 
+type testCloudModules struct {
+	modules.Modules
+}
+
+func (m *testCloudModules) Features() modules.Features {
+	return modules.Features{
+		Cloud: true, // Explicily turn on cloud feature.
+	}
+}
+
+// TestReceivingAccountRecoveryCodesWhenChangingAuthns tests for following:
+//  - Recovery codes are not returned for usernames that are not emails
+//  - Recovery codes are returned for usernames that are valid emails
+func TestReceivingAccountRecoveryCodesWhenChangingAuthenticationsWithToken(t *testing.T) {
+	env := newWebPack(t, 1)
+	ctx := context.Background()
+
+	// Enable second factor.
+	ap, err := types.NewAuthPreference(types.AuthPreferenceSpecV2{
+		Type:         constants.Local,
+		SecondFactor: constants.SecondFactorOTP,
+	})
+	require.NoError(t, err)
+	err = env.server.Auth().SetAuthPreference(ctx, ap)
+	require.NoError(t, err)
+
+	// Enable cloud feature.
+	defaultModules := modules.GetModules()
+	defer modules.SetModules(defaultModules)
+	modules.SetModules(&testCloudModules{})
+
+	// Creaet a username that is not a valid email format for recovery.
+	teleUser, err := types.NewUser("invalid-name-for-recovery")
+	require.NoError(t, err)
+	env.server.Auth().CreateUser(ctx, teleUser)
+
+	// Create a reset password token and secrets.
+	token, err := env.server.Auth().CreateResetPasswordToken(ctx, auth.CreateUserTokenRequest{
```

Comment:
Rename `token` to `resetToken`. Let's be specific about which token we are using when we can.